pkg/operator/ceph/cluster/cluster_external.go

```diff
@@ -199,18 +208,14 @@ func (c *ClusterController) configureExternalClusterMonitoring(cluster *cluster)
 	// Create external monitoring Service
 	service := manager.MakeMetricsService(mgr.ExternalMgrAppName, mgr.ServiceExternalMetricName)
 	logger.Info("creating mgr external monitoring service")
-	_, err := c.context.Clientset.CoreV1().Services(c.namespacedName.Namespace).Create(ctx, service, metav1.CreateOptions{})
-	if err != nil {
-		if !kerrors.IsAlreadyExists(err) {
-			return errors.Wrap(err, "failed to create mgr service")
-		}
-		logger.Debug("mgr external metrics service already exists")
-	} else {
-		logger.Info("mgr external metrics service created")
+	_, err := k8sutil.CreateOrUpdateService(c.context.Clientset, cluster.Namespace, service)
```

Comment:
How about we move this code block inside the `mgr` project? Then we don't have to expose details like the default prometheus port outside the mgr project.

Reply:
The port is used in the `validateExternalClusterSpec()` function only. I'm just not sure how we can move this block in `mgr` and making things simpler. Also I'd like to keep this change as simple as possible.